### ISSUE TITLE
Sample Files don't export with Midis

### ISSUE THREAD

**Issue by ElectricBee1** [LMMS/lmms#4185]
Hello! For some reason, midi samples (for instruments) don't export when I export a song that I use a sf2 soundfont with. In other words, everything defaults to a piano when I export my song as a MIDI. Is there a workaround until sample exporting becomes native to LMMS?

**Comment by tresf** [comment]
As far as I know MIDI export simply exports a plain "MIDI" file with any specific "instrument type" information.  This means that the instrument mapping MIDI composers would be use to are never going to make it in properly and this is by design since we wouldn't really know.  MIDI is designed more as a way to export note information.  MIDI export is not intended to work properly out of the box with a particular MIDI playback software.

To elaborate... in order for LMMS to produce get generic MIDI instrument sounds with the exported file, we would have to categorize every SF2 and preset to match a particular instrument category.  This won't happen.  Please correct me if any part of this is wrong.

**Comment by Anonymouqs** [comment]
The only standard you could really use is the GM standard, but even there, it can get pretty hectic in percussion instruments. (There's no standard encoding for Snare except a regular stroke in GM)

I have been able to export midi **from** MuseScore 2.0 and into LMMS and have it play it correctly with each of the instruments.

@ElectricBee1  Please give steps to reproduce, I don't know if you're talking about export from LMMS or from another program.

**Comment by Umcaruje** [comment]
>we would have to categorize every SF2 and preset to match a particular instrument category. This won't happen. Devs, please correct me if any part of this is wrong.

Nope, you have a standardized MIDI sound bank across all manufacturers to ensure all files are played properly. This is what's known as the [General MIDI Standard](https://en.wikipedia.org/wiki/General_MIDI) So I think we should be exposing a way for the users to set channels on their midi tracks as well as a desired patch number for the instrument.

**Comment by Anonymouqs** [comment]
> So I think we should be exposing a way for the users to set channels on their midi tracks as well as a desired patch number for the instrument.

I feel like this is going to end up becoming an enhancement thread so I'm going to apply the enhancement label. On the topic of setting channels and patch numbers, I think that would be really helpful in importing midis from Musescore 2 and other Notation Software.
